device/sycl/include/traccc/sycl/clusterization/measurement_creation.hpp

```diff
@@ -0,0 +1,63 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// SYCL library include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/edm/cell.hpp"
+#include "traccc/edm/cluster.hpp"
+#include "traccc/edm/measurement.hpp"
+#include "traccc/sycl/utils/queue_wrapper.hpp"
+#include "traccc/utils/algorithm.hpp"
+
+namespace traccc::sycl {
+
+struct measurement_creation
+    : public algorithm<host_measurement_container(
+          const host_cluster_container &, const std::vector<std::size_t> &,
```

Comment:
It seems redundant to pass both the cluster container and a vector describing the sizes of the cluster container as input to this algorithm. The information stored in the `std::vector<std::size_t>` variable is available from the `traccc::host_cluster_container` object as well. Passing it separately just gives a chance for passing incorrect values to the algorithm.

Reply:
Alright, this is done too.